python/source/org/libreoffice/ide/eclipse/python/PythonMainProvider.java

```diff
@@ -1,149 +0,0 @@
- */
-public class PythonMainProvider implements IMainProvider {
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public Vector<String> getMainNames(IProject pProject) {
-        Vector<String> mains = new Vector<String>();
-
-        IJavaProject javaPrj = JavaCore.create(pProject);
-        try {
-            mains.addAll(getInternalMainNames(javaPrj));
-        } catch (Exception e) {
-        }
-
-        return mains;
-    }
-
-    /**
-     * Recursive method to find the Classes and check their hierarchy.
-     *
-     * @param pElement the Java AST element for scan for XMain implementations
-     * @return the names of the classes implementing the XMain interface in the
-     *      Java AST element.
-     */
-    private Vector<String> getInternalMainNames(IParent pElement) {
-        Vector<String> mains = new Vector<String>();
-
-        try {
-            for (IJavaElement child : pElement.getChildren()) {
-
-                boolean visit = true;
-
-                if (child instanceof IPackageFragmentRoot) {
-                    IPackageFragmentRoot root = (IPackageFragmentRoot) child;
-                    if (root.getKind() != IPackageFragmentRoot.K_SOURCE) {
-                        visit = false;
-                    }
-                }
-
-                if (visit) {
-                    if (child instanceof ICompilationUnit) {
-                        ICompilationUnit unit = (ICompilationUnit) child;
-                        IType type = unit.findPrimaryType();
-
-                        if (isMainImplementation(type)) {
-                            mains.add(type.getFullyQualifiedName());
-                        }
-                    } else if (child instanceof IParent) {
-                        mains.addAll(getInternalMainNames((IParent) child));
-                    }
-                }
-            }
-        } catch (Exception e) {
-        }
-
-        return mains;
-    }
-
-    /**
-     * Checks if the Java type implements the <code>com.sun.star.lang.XMain</code>
-     * interface.
-     *
-     * @param pType the Java type to check
-     * @return <code>true</code> if the type implements <code>XMain</code>,
-     *      <code>false</code> otherwise.
-     */
-    private boolean isMainImplementation(IType pType) {
-        boolean isMainImplementation = false;
-
-        try {
-            ITypeHierarchy hierarchy = pType.newSupertypeHierarchy(null);
-            IType[] superInterfaces = hierarchy.getAllSuperInterfaces(pType);
-
-            int i = 0;
-            while (!isMainImplementation && i < superInterfaces.length) {
-                if (superInterfaces[i].getFullyQualifiedName().equals("com.sun.star.lang.XMain")) { //$NON-NLS-1$
-                    isMainImplementation = true;
-                } else {
-                    i++;
-                }
-            }
-        } catch (Exception e) {
-        }
-
-        return isMainImplementation;
-    }
-}
```

Comment:
PythonMainProvider: This class was removed and also its corresponding entry in the plugin.xml
The requirement of the corresponding JavaMainProvider's is to provide the main class for the running of the URE application. Now we currently do not have any such thing as for the Python Extension, so removing the PythonMainProvider.